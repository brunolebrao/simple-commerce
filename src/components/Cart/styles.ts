import styled from 'styled-components'

export const Container = styled.aside`
  width: 500px;
  padding: 20px;
`
