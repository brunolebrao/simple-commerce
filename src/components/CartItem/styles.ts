import styled from 'styled-components'

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid ${(props) => props.theme.colors.primary.main};
  padding: 15px 0;
  font-size: 1.3rem;
  height: 130px;
`
export const Content = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  width: 100%;
`
export const Values = styled.div`
  display: flex;
  justify-content: space-between;
  font-size: 1.5rem;
`
export const Information = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`
export const Amount = styled.p`
  margin: 0 1rem;
`
export const Img = styled.img`
  max-width: 80px;
  object-fit: cover;
  margin-left: 40px;
`
