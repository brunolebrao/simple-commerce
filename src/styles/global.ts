import { createGlobalStyle } from 'styled-components'

const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  html {
  }

  html, body, #__next {
    height: 100%;
  }

  body {
    -webkit-font-smoothing: antialiased;
  }
`
export default GlobalStyle
