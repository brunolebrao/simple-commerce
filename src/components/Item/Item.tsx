import * as S from './styles'
import { ProductTypes } from 'types/products.types'
import React from 'react'

interface ItemProps {
  item: ProductTypes
  handleAddToCart: (clickedItem: ProductTypes) => void
}

const Item = ({ item, handleAddToCart }: ItemProps) => (
  <S.Container>
    <S.Img src={item.image} alt={item.title} />
    <S.Article>
      <h3>{item.title}</h3>
      <p>{item.description}</p>
      <h3>R${item.price}</h3>
    </S.Article>
    <S.CustomButton
      color="primary"
      variant="contained"
      onClick={() => handleAddToCart(item)}
    >
      Adicionar no carrinho
    </S.CustomButton>
  </S.Container>
)

export default Item
