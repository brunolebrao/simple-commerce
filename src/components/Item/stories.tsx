import { Story, Meta } from '@storybook/react/types-6-0'
import Item from '.'

export default {
  title: 'Item',
  component: Item
} as Meta

export const Basic: Story = () => <Item />
