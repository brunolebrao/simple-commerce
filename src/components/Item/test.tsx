import { render } from '@testing-library/react'
import { ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../styles/theme'
import Item from '.'

const props = {
  item: {
    category: "men's clothing",
    description:
      'Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday',
    id: 1,
    image: 'https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg',
    price: 109.95,
    title: 'Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops',
    amount: 0
  },
  handleAddToCart: jest.fn()
}

const renderComponent = (children: ReactNode) =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

describe('<Item />', () => {
  it('Should render the heading', () => {
    const { container } = renderComponent(<Item {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
