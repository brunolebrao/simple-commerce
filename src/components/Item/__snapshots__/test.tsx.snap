// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> Should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  border: 1px solid #06092b;
  border-radius: 20px;
  height: 100%;
}

.c1 {
  max-height: 250px;
  object-fit: cover;
  border-radius: 20px 20px 0 0;
}

.c2 {
  padding: 1rem;
  height: 100%;
}

.c3 {
  border-radius: 0 0 20px 20px;
  font-size: 1rem;
}

<main
  class="c0"
>
  <img
    alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
    class="c1"
    src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
  />
  <article
    class="c2"
  >
    <h3>
      Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
    </h3>
    <p>
      Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
    </p>
    <h3>
      R$
      109.95
    </h3>
  </article>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained c3 MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Adicionar no carrinho
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</main>
`;
